var/spack/repos/builtin/packages/npm/package.py

```diff
@@ -9,13 +9,18 @@
 from spack.package import *
 
 
-# NOTE: not actually an Autotools package
 class Npm(Package):
     """npm: A package manager for javascript."""
 
     homepage = "https://github.com/npm/cli"
     # base https://www.npmjs.com/
-    url = "https://registry.npmjs.org/npm/-/npm-6.13.4.tgz"
+
+    git = "https://github.com/npm/cli.git"
+    url = "https://registry.npmjs.org/npm/-/npm-9.30.0.tgz"
+
+    version("9.30.0", commit="4d275927e069606c8047b678ad09aafbe18b1673")
+    version("8.19.3", commit="ef4299658330470b56966b48394824895b2d6d04")
+    version("7.24.2", commit="04eb43f2b2a387987b61a7318908cf18f03d97e0")
 
     version("6.14.9", sha256="1e0e880ce0d5adf0120fb3f92fc8e5ea5bac73681d37282615d074ff670f7703")
```

Comment:
Thoughts on deprecating and eventually removing older versions? There are currently only 2 packages in Spack that depend on npm, suggesting it isn't very popular in the HPC community. These older versions significantly increase the complexity of the build recipe. I'm not sure if anyone actually needs npm 3 or 6 anymore, or even 7 and 8...

Reply:
Maybe throw a warning when people try to use "deprecated" versions, but keep them in the recipe so reproducibility isn't impacted?

Reply:
Marking them as deprecated will throw a warning automatically. But the only point of deprecating them is to someday remove them.